Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -1088,16 +1088,18 @@ ALWAYS_INLINE static void do_draw_scaled_bitmap(Gfx::Bitmap& target, const IntRe
     }
 
     bool has_opacity = opacity != 1.0f;
-    int hscale = (src_rect.width() * (1 << 16)) / dst_rect.width();
-    int vscale = (src_rect.height() * (1 << 16)) / dst_rect.height();
-    int src_left = src_rect.left() * (1 << 16);
-    int src_top = src_rect.top() * (1 << 16);
+    // Assuming long long int's are 64 bits wide.
+    long long int shift = (long long int)1 << 32;
```

Comment:
What about using the `i64` type instead? That way we don't have to assume that a `long long int` is 64 bit, but instead the type exactly tells us how many bits are used :^)

Reply:
That's too easy! :^)
I'll do that, thanks